### DISCUSSION TITLE
aws sdk 1.12.x How to using IMDSv2

### DISCUSSION THREAD

**Discussion by Navapon** [aws/aws-sdk-java#3081] [Q&A]
Hello,

I would like to know how we config using AWS SDK version 1.12.x using ec2 metadata IMDSv2 

I thought default using v1 because I changed EC2 instance to only support IMDSv2 my application got an error 

```
2024-01-17T10:20:44.152Z LOG_TYPE=LIB_LOG [main] WARN com.amazonaws.util.EC2MetadataUtils.getItems - Unable to retrieve the requested metadata (/latest/dynamic/instance-identity/document). Unauthorized (Service: null; Status Code: 401; Error Code: null; Request ID: null; Proxy: null)
```

Does anyone have a guide or recommendation please help 

Thank you in advance

**Comment by bhoradc** [comment]
Hi @Navapon,

IMDSv2 was introduced in Java SDK version 1.11.678 onwards. Since you are on SDK version 1.12.x, IMDSv2 is used by default. In the scenario, when the session token is not found for the request, it will try to fallback to IMDSv1.

In your case, for IMDSv2, the HTTP error - `401 - Unauthorized` is returned when the GET request uses an invalid token. Did you get a chance to validate the token or regenerating a new one? 

You may also want to refer to this [developer guide reference](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/migration-imds.html) for usage of EC2 metadata utility in Java SDK V1. 

Regards, 
Chaitanya

**Reply by Navapon** [reply]
Hi @bhoradc 

Thank you for your responding

I appreciate your recommendation. I will try to follow it.
